src/selectors/index.ts

```diff
@@ -67,3 +81,46 @@ export const selectSchemaFieldDefs = createSelector(
     });
   }
 );
+
+const selectResult = (state: State) => state.present.result.main;
+
+const selectMainSpec = createSelector(
+  selectData,
+  selectFilters,
+  selectResult,
+  (data: Data, filters: RangeFilter[] | OneOfFilter[], result: Result) => {
+    if (!result.modelGroup) {
+      return;
+    }
+    return {
+      data: data,
+      transform: getTransforms(filters),
+      ...getTopSpecQueryItem(result.modelGroup).spec
+    };
+  }
+);
+
+export const selectMainResult = createSelector(
```

Comment:
name this `selectMainResultForView` (so `selectResult` can be `selectMainResult` like before?)